server/utils/urlScraping.js

```diff
@@ -7,20 +7,24 @@ const scrapingAmazon = async (url) => {
   const browser = await puppeteer.launch();
   const page = await browser.newPage();
   page.setUserAgent(
-    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36",
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
   );
   // Go to Amazon Product page
   try {
     await page.goto(url, {
       waitUntil: ["load", "domcontentloaded", "networkidle0", "networkidle2"],
     });
     await page.waitForSelector("#productTitle");
-    await page.waitForSelector("#priceblock_ourprice");
 
     // Extract information from page
     const { title, price, imageUrl } = await page.evaluate(() => {
       const title = document.getElementById("productTitle").innerText;
-      const price = document.getElementById("priceblock_ourprice").innerText;
+      let price = null;
+      if (document.getElementById("priceblock_ourprice")) {
+        price = document.getElementById("priceblock_ourprice").innerText;
+      } else if (document.getElementById("priceblock_saleprice")) {
+        price = document.getElementById("priceblock_ourprice").innerText;
+      }
```

Comment:
Should there be different logic inside these two blocks

Reply:
yes, line 26 should be 'priceblock_saleprice'